From 36167a7b6617b0c8e1e15a91eccf5cdb19f0ee1b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lu=C3=ADs=20Ferreira?= <contact@lsferreira.net>
Date: Sun, 19 May 2019 19:22:41 +0100
Subject: [PATCH] hwdb: update ACCEL_LOCATION property to use Or instead of
 QuotedString

(cherry picked from commit 331e34fe1db427127aefa3b5ca6d5b568ecf86d2)

Related: RHEL-130979
---
 hwdb/parse_hwdb.py | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/hwdb/parse_hwdb.py b/hwdb/parse_hwdb.py
index 6943c688eb..99fb5b1438 100755
--- a/hwdb/parse_hwdb.py
+++ b/hwdb/parse_hwdb.py
@@ -128,7 +128,7 @@ def property_grammar():
              ('KEYBOARD_LED_NUMLOCK', Literal('0')),
              ('KEYBOARD_LED_CAPSLOCK', Literal('0')),
              ('ACCEL_MOUNT_MATRIX', mount_matrix),
-             ('ACCEL_LOCATION', STRING),
+             ('ACCEL_LOCATION', Or(('display', 'base'))),
             )
     fixed_props = [Literal(name)('NAME') - Suppress('=') - val('VALUE')
                    for name, val in props]
@@ -180,10 +180,6 @@ def check_one_default(prop, settings):
     if len(defaults) > 1:
         error('More than one star entry: {!r}', prop)
 
-def check_one_accel_location(prop, value):
-    if value not in ['base', 'display']:
-        error('Wrong accel location: {!r}', prop)
-
 def check_one_mount_matrix(prop, value):
     numbers = [s for s in value if s not in {';', ','}]
     if len(numbers) != 9:
@@ -226,8 +222,6 @@ def check_properties(groups):
                 check_one_default(prop, parsed.VALUE.SETTINGS)
             elif parsed.NAME == 'ACCEL_MOUNT_MATRIX':
                 check_one_mount_matrix(prop, parsed.VALUE)
-            elif parsed.NAME == 'ACCEL_LOCATION':
-                check_one_accel_location(prop, parsed.VALUE)
             elif parsed.NAME.startswith('KEYBOARD_KEY_'):
                 check_one_keycode(prop, parsed.VALUE)
 
